### ISSUE TITLE
Hyper-v in Win10 Failed to insert module 'intel_sgx':no such device

### ISSUE THREAD

**Issue by yanzhichao** [openenclave/openenclave#4262]
### description
I try to create a vm with sgx use Hyper-v, I follows HyperVLinuxVMSetup.md: https://github.com/openenclave/openenclave/blob/master/docs/GettingStartedDocs/HyperVLinuxVMSetup.md

but it don't work, with the error as below:

![1632482129(1)](https://user-images.githubusercontent.com/16754580/134666035-c6002ea3-c3c9-457a-b456-b774acfc2ced.png)

My VM OS is ubuntu20.04 server  with linux-kernel 5.4
and the version of the sgx driver is sgx_linux_x64_driver_1.36.2.bin for ubuntu20.04

Host OS is windows10 professional  
CPU is Intel core i7-10510 which support sgx without FLC 

Did I miss something ? Could you give me some suggestion ?

**Comment by anakrish** [comment]
@yanzhichao 
Can you confirm if:
1. SecureBoot has been disabled for the VM.
2. `Set-VMSgx -VmName MyVM -SgxEnabled $True -SgxSize 32` has successfully executed.

**Comment by yanzhichao** [comment]
Yes, secureboot is unset, and set-vmsgx  has successfully as below:
![image](https://user-images.githubusercontent.com/16754580/134746697-1aa1df25-d61b-45a7-bdeb-4b9d5164c68e.png)

**Comment by anakrish** [comment]
On your VM, from within the OE SDK root folder, do:
```bash
sudo ./scripts/ansible/install-ansible.sh
sudo ansible-playbook ./scripts/ansible/oe-contributors-setup-sgx1.yml
```

You ought to then see `/dev/isgx` instead of `/dev/intel_sgx`

**Comment by yanzhichao** [comment]
Yes, not use DCAP driver can work well.  Actually, I want to use DCAP driver for remote attestion. I know that  DCAP need cpu support FLC. so I change a cpu ( Intel(R) Pentium(R) Silver J5005 ) which support FLC, and have a try,  it don't work too. Also it can Set-VMSgx -VmName MyVM -SgxEnabled $True -SgxSize 32 successful.  but the FLC(SGX_LC: SGX launch config supported = false) is  flase. 
Is DCAP not supported yet in VM ?

- Set-VmSgx

![image](https://user-images.githubusercontent.com/16754580/135011284-685ab3c5-c9ca-4e34-b8e3-27747089f28c.png)

- the infomation in vm:

![image](https://user-images.githubusercontent.com/16754580/135011269-bf687f24-10c3-4434-a898-23dac7618edc.png)

**Comment by anakrish** [comment]
@radhikaj @yentsanglee @mingweishih Can you chime in?

**Comment by haitaohuang** [comment]
Looks like your VM is not setup correctly as cpuid reports no FLC support.
Also make sure in BIOS FLC is enabled. Usually an entry like "SGXLEPUBKEYHASHx Write {Enable, Disable}" needs set to enable

(After vm/bios setup correctly, also note for Ubuntu 20.04, if you are manually installing DCAP driver without installing libsgx-enclave-common, you may need remount /dev without noexec flag)

**Comment by yanzhichao** [comment]
I make it work .
There are two additional parameters in `Set-VMSgx`  : 
```
        # Desired default SGX Launch Control Mode for the VM.
        [Parameter(Position = 3)]
        [String]$SgxLaunchControlDefault,

        # Desired SGX Launch Control Mode for the VM.
        [Parameter(Position = 4)]
        [UInt32]$SgxLaunchControlMode

```
the default `SgxLaunchControlMode` is 0,  I try to set `SgxLaunchControlMode`  to 1 and 2 , when I set 1, the cpuid shows FLC is true. But it still failed to insert moulde intel_sgx.  

log is as below:
![image](https://user-images.githubusercontent.com/16754580/135198361-13195476-8b14-4b00-ab0d-ebb451ef9b73.png)

![image](https://user-images.githubusercontent.com/16754580/135198560-45e27508-d75a-4102-8dca-2584b5f0d2c6.png)

but when I try to set the SgxLaunchControlMode to 2 ， it works：
![image](https://user-images.githubusercontent.com/16754580/135198669-7197adc1-908b-47de-ac11-5bf1cf297276.png)

![image](https://user-images.githubusercontent.com/16754580/135198750-fc38c31b-d258-420b-bdb3-2385ee851e10.png)

Who knows the role of these two parameters ：`SgxLaunchControlMode` and `SgxLaunchControlDefault` .  And How to configure them ?

**Comment by anakrish** [comment]
@HernanGatta Would be the best person for `Set-VmSgx`

**Comment by radhikaj** [comment]
Here is some information we have received. 

SgxLaunchControlMode: Flexible launch control (FLC) mode allows the guest's operating system to control which enclaves to be launched on the host. The specified MSR here is IA32_SGXLEPUBKEYHASH*.

0 – NoAccess: The FLC MSRs are not exposed to the guest at all
1 – ReadOnly: The FLC MSRs are only readable but not writable from the guest
2 – ReadWrite: The FLC MSRs are read-writeable for the guest

SgxLaunchControlDefault specified the default SGX launch control MSR values. Each 64-bit MSR value must be specified as a 16 character little-endian hexadecimal number.
